test/Catch.hs

```diff
@@ -11,26 +11,24 @@ import Test.Tasty.Hedgehog
 
 gen
   :: forall e m a sig
-  .  (Has (Catch e) sig m, Arg e, Show a, Show e, Vary e)
+  .  (Has (Catch e) sig m, Arg e, Show e, Vary e)
   => Gen e
-  -> (forall a . Show a => Gen a -> Gen (With (m a)))
+  -> (forall a . Gen a -> Gen (m a))
   -> Gen a
-  -> Gen (With (m a))
-gen _ m a = do
-  h <- fn @e (m a)
-  subterm (m a) $ \ m -> liftWith2 "catchError" catchError m (fmap getWith <$> showingFn h)
+  -> Gen (m a)
```

Comment:
1. We no longer require `Show` constraints.

2. We no longer require `With`.

Both are due to the fact that `Gen` is now a monad we construct wrapping `Hedgehog.Gen (Term a)` (`Term` being a deep embedding of the prior `With` logic).